### ISSUE TITLE
Model contains Document's properties

### ISSUE THREAD

**Issue by googlicius** [typegoose/typegoose#220]
## Versions

- NodeJS: v10.19.0
- Typegoose(NPM): ^6.4.0 
- Typegoose(GIT): commithash
- mongoose: ^5.9.7
- mongodb: 0.0.0
- type-graphql: ^0.17.6

## What is the Problem?
Currently, Document's properties available in Model, but it's shouldn't.

## Code Example

```ts
export class User {
  @prop({ required: true })
  firstName!: string;

  @prop({ required: true })
  lastName!: string;
}

// Create user model
const UserModel = getModelForClass(User);

// vscode intellisense tell me that firstName is a property of UserModel.

UserModel.firstName // Got undefined.
```

## Do you know *why* it happenes?

I think ReturnModelType has extracted all Document's properties into Model.

**Comment by hasezoey** [comment]
this is already known, but i didnt find a way to exclude these while also still being compatible with other types (im not so good with typescript types)

**Comment by googlicius** [comment]
Let me try to find a solution for this.

**Comment by googlicius** [comment]
@hasezoey I have check and addressed some issues: 

In `src/types.ts`: 
```ts
export type ModelType<T> = mongoose.Model<DocumentType<T>> & T;
```
ModelType is inconsistent with mongoose.Model type, mongoose.Model has 2 generics types: Document and QueryHelpers. But Typegoose has only one. With this missing, we cannot add more methods to model which 3rd party plugins usually do.

And we cannot using logical AND `& T` here as well, so it will public all document's properties to model.

**Comment by hasezoey** [comment]
should be fixed since #227
